### ISSUE TITLE
In WSL the init command doesn't run properly on shell startup.

### ISSUE THREAD

**Issue by Jonxslays** [MrDogeBro/quicknav#47]
## Bug Description

After installing quicknav, adding the init command to the `.bashrc`, and restarting the terminal or sourcing from the `.bashrc` a warning is emitted that quicknav is not found.

## Reproduction
1) Install Quicknav inside WSL.
2) Add the `eval "$(quicknav init bash)"` to your `.bashrc`.
3) Close the WSL terminal, and open a new one.
4) You will see the message `-bash: quicknav: command not found`.
5) Manually run  `eval "$(quicknav init bash)"` and quicknav gets initialized properly.

## Expected Behavior
I would expect quicknav to init properly, and allow the usage of the nav command. Is there anyway to bundle in the quicknav init stuff with the installation of quicknav, removing the need to add it to the .rc files altogether? Maybe utilizing the `$SHELL` environment variable and cargo workspaces so that nav is also set as a command immediately?

## Screenshots
![image](https://user-images.githubusercontent.com/51417989/143484541-528f63b1-6401-42d8-bdc3-f1928e297f6d.png)

## Info
- OS Name and Version: [e.g. OSX 11.0.0] Debian GNU/Linux 11 (bullseye)
- Quicknav Version [e.g. 1.0.0] 1.3.0
- Shell: [e.g. Zsh] Bash

## Additional context

I don't know precisely what the cause of the issue is. I have only seen it on WSL. Works perfect on my Arch system. I saw someone else have this issue as well (also WSL), so I figured it was worth bringing up.

**Comment by Jonxslays** [comment]
I just remembered the nav command name is variable and can be changed by the user. So making it init the command upon install instead of shell startup would likely cause issues with that. I'm curious what your thoughts on this are.

**Comment by MrDogeBro** [comment]
Hmmm.......I'm confused. That doesn't make any sense why it would do so. It almost seems that its loading the `~/.bashrc` file before it actually loads the programs that are installed if that makes sense. That's very weird....

**Comment by Jonxslays** [comment]
Yeah i also thought it was very odd. I will try to do some testing in a bit to see if I can find a fix from the wsl end.

**Comment by MrDogeBro** [comment]
Ok...it just seems very weird as if something is loading wrong....

**Comment by parafoxia** [comment]
Does it happen when loading a new terminal, or when loading the first terminal after WSL starts up?

**Comment by Jonxslays** [comment]
After investigation into [this issue](https://github.com/Microsoft/WSL/issues/776) regarding wsl loading too slowly, I got the idea to create a function that check if quicknav has been initialized and if not does so, the first time the quicknav command is run in a session. It may lead to unnecessary quicknav initializations, but that is not an issue for me, and it resolves the problem completely.

When quicknav is initialized any time *after* wsl startup it works fine.

Bottom line: This is not a quicknav issue, it is a wsl issue.


```bash
# ~/.bashrc

export IS_QN_INIT=0

initialize_quicknav() {
    if [ $IS_QN_INIT -eq 0 ]; then
        alias quicknav=quicknav
        eval "$(quicknav init bash)"
        export IS_QN_INIT=1
    fi
}

alias quicknav="initialize_quicknav; quicknav"
```
